### ISSUE TITLE
add interval argument to expect_row_values_to_have_data_for_every_n_datepart

### ISSUE THREAD

**Issue by lewisarmistead** [calogica/dbt-expectations#109]
We manage large amounts of power industry-related time series data and need to validate presence of data at a granularity between minutely and hourly, e.g. we expect many of our tables to have data at every 5-minute interval.

I see that the [`expect_row_values_to_have_data_for_every_n_datepart`](https://github.com/calogica/dbt-expectations/blob/main/macros/schema_tests/distributional/expect_row_values_to_have_data_for_every_n_datepart.sql) test uses the [`get_base_dates`](https://github.com/calogica/dbt-date/blob/main/macros/get_base_dates.sql) macro from dbt-date to create the date spine used to identify missing data; it would be really helpful to have a line in the [`base_dates`](https://github.com/calogica/dbt-expectations/blob/74d0c218a4fbd7f701420d2b2fba3cef15a70147/macros/schema_tests/distributional/expect_row_values_to_have_data_for_every_n_datepart.sql#L42-L46) CTE to filter the spine to return only X-date_part intervals where X is the value of a new argument - something like this where I used `every_n_datepart` as the new arg:

```sql
{% set start_date = '2021-09-14' %}
{% set end_date = '2021-09-15' %}
{% set date_part = 'minute' %}
{% set every_n_datepart = 5 %}

with base_dates as (

    {{dbt_date.get_base_dates(start_date=start_date, end_date=end_date, datepart=date_part)}}
    where mod({{ dbt_date.date_part(date_part, 'date_' + date_part) }}, {{every_n_datepart}}) = 0

),
```

This addition would allow us to check only for presence at the 5-minute interval in this example (at least when I ran it on Snowflake).

The default for this new datepart interval argument could be `None` with a condition for no-op similar to how some of the other arguments are handled:

```sql
with base_dates as (

    {{dbt_date.get_base_dates(start_date=start_date, end_date=end_date, datepart=date_part)}}
    {% if every_n_datepart %}
    where mod({{ dbt_date.date_part(date_part, 'date_' + date_part) }}, {{every_n_datepart}}) = 0
    {% endif %}

),
```

This may better fit as a feature of dbt-date or the dbt_utils.date_spine macro, but this is where we need it most right now.

Regardless of whether this feature gets implemented or not, thanks for the awesome package!

**Comment by clausherther** [comment]
Hi @lewisarmistead! Thanks for this issue, I'll take a look by this weekend. Would you interested in tackling this in a PR?
One thing that jumps out right away is whether `mod` works across platforms or may need to be wrapped in an adapter macro.

**Comment by lewisarmistead** [comment]
Hi @clausherther - I'd be happy to throw this into a PR. Thanks for the note on the `mod` operator - I'll see if I can come up with something that'll work across platforms.

**Comment by clausherther** [comment]
Closed via #110
